docs/rules/prefer-mock-promise-shorthand.md

```diff
@@ -0,0 +1,34 @@
+# Prefer mock resolved/rejected shorthands for promises (`prefer-mock-promise-shorthand`)
+
+When working with mocks of functions that return promises, Jest provides some
+syntactic sugar functions to reduce the amount of boilerplate you have to write.
```

Comment:
it's not really syntax, it's API

Reply:
I took that phrasing from [the jest docs](https://jestjs.io/docs/mock-function-api#mockfnmockreturnthis) 😅

Reply:
Done: https://github.com/facebook/jest/pull/13080